doc/man/gobyted.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.48.3.
.TH GOBYTED "1" "May 2021" "gobyted v0.17.0.3" "User Commands"
.SH NAME
gobyted \- manual page for gobyted v0.17.0.3
.SH DESCRIPTION
GoByte Core Daemon version v0.17.0.3
.SS "Usage:"
.TP
gobyted [options]
Start GoByte Core Daemon
.SH OPTIONS
.HP
\-?
.IP
Print this help message and exit
.HP
\fB\-alertnotify=\fR<cmd>
.IP
Execute command when a relevant alert is received or we see a really
long fork (%s in cmd is replaced by message)
.HP
\fB\-assumevalid=\fR<hex>
.IP
If this block is in the chain assume that it and its ancestors are valid
and potentially skip their script verification (0 to verify all,
default:
00000000000000105cfae44a995332d8ec256850ea33a1f7b700474e3dad82bc,
testnet:
0000009303aeadf8cf3812f5c869691dbd4cb118ad20e9bf553be434bafe6a52)
.HP
\fB\-blocknotify=\fR<cmd>
.IP
Execute command when the best block changes (%s in cmd is replaced by
block hash)
.HP
\fB\-blockreconstructionextratxn=\fR<n>
.IP
Extra transactions to keep in memory for compact block reconstructions
(default: 100)
.HP
\fB\-blocksdir=\fR<dir>
.IP
Specify blocks directory (default: <datadir>/blocks)
.HP
\fB\-conf=\fR<file>
.IP
Specify configuration file. Relative paths will be prefixed by datadir
location. (default: gobyte.conf)
.HP
\fB\-daemon\fR
.IP
Run in the background as a daemon and accept commands
.HP
\fB\-datadir=\fR<dir>
.IP
Specify data directory
.HP
\fB\-dbcache=\fR<n>
.IP
Set database cache size in megabytes (4 to 16384, default: 300)
.HP
\fB\-debuglogfile=\fR<file>
.IP
Specify location of debug log file. Relative paths will be prefixed by a
net\-specific datadir location. (0 to disable; default: debug.log)
.HP
\fB\-loadblock=\fR<file>
.IP
Imports blocks from external blk000??.dat file on startup
.HP
\fB\-maxmempool=\fR<n>
.IP
Keep the transaction memory pool below <n> megabytes (default: 300)
.HP
\fB\-maxorphantxsize=\fR<n>
.IP
Maximum total size of all orphan transactions in megabytes (default: 10)
.HP
\fB\-maxrecsigsage=\fR<n>
.IP
Number of seconds to keep LLMQ recovery sigs (default: 604800)
.HP
\fB\-mempoolexpiry=\fR<n>
.IP
Do not keep transactions in the mempool longer than <n> hours (default:
336)
.HP
\fB\-par=\fR<n>
.IP
Set the number of script verification threads (\fB\-4\fR to 16, 0 = auto, <0 =
leave that many cores free, default: 0)
.HP
\fB\-persistmempool\fR
.IP
Whether to save the mempool on shutdown and load on restart (default: 1)
.HP
\fB\-pid=\fR<file>
.IP
Specify pid file. Relative paths will be prefixed by a net\-specific
datadir location. (default: gobyted.pid)
.HP
\fB\-prune=\fR<n>
.IP
Reduce storage requirements by enabling pruning (deleting) of old
blocks. This allows the pruneblockchain RPC to be called to
delete specific blocks, and enables automatic pruning of old
blocks if a target size in MiB is provided. This mode is
incompatible with \fB\-txindex\fR, \fB\-rescan\fR and \fB\-disablegovernance\fR=\fI\,false\/\fR.
Warning: Reverting this setting requires re\-downloading the
entire blockchain. (default: 0 = disable pruning blocks, 1 =
allow manual pruning via RPC, >945 = automatically prune block
files to stay under the specified target size in MiB)
.HP
\fB\-syncmempool\fR
.IP
Sync mempool from other nodes on start (default: 1)
.HP
\fB\-sysperms\fR
.IP
Create new files with system default permissions, instead of umask 077
(only effective with disabled wallet functionality)
.HP
\fB\-version\fR
.IP
Print version and exit
.PP
Connection options:
.HP
\fB\-addnode=\fR<ip>
.IP
Add a node to connect to and attempt to keep the connection open (see
the `addnode` RPC command help for more info). This option can be
specified multiple times to add multiple nodes.
.HP
\fB\-allowprivatenet\fR
.IP
Allow RFC1918 addresses to be relayed and connected to (default: 0)
.HP
\fB\-banscore=\fR<n>
.IP
Threshold for disconnecting misbehaving peers (default: 100)
.HP
\fB\-bantime=\fR<n>
.IP
Number of seconds to keep misbehaving peers from reconnecting (default:
86400)
.HP
\fB\-bind=\fR<addr>
.IP
Bind to given address and always listen on it. Use [host]:port notation
for IPv6
.HP
\fB\-connect=\fR<ip>
.IP
Connect only to the specified node; \fB\-connect\fR=\fI\,0\/\fR disables automatic
connections (the rules for this peer are the same as for
\fB\-addnode\fR). This option can be specified multiple times to connect
to multiple nodes.
.HP
\fB\-discover\fR
.IP
Discover own IP addresses (default: 1 when listening and no \fB\-externalip\fR
or \fB\-proxy\fR)
.HP
\fB\-dns\fR
.IP
Allow DNS lookups for \fB\-addnode\fR, \fB\-seednode\fR and \fB\-connect\fR (default: 1)
.HP
\fB\-dnsseed\fR
.IP
Query for peer addresses via DNS lookup, if low on addresses (default: 1
unless \fB\-connect\fR used)
.HP
\fB\-enablebip61\fR
.IP
Send reject messages per BIP61 (default: 1)
.HP
\fB\-externalip=\fR<ip>
.IP
Specify your own public address
.HP
\fB\-forcednsseed\fR
.IP
Always query for peer addresses via DNS lookup (default: 0)
.HP
\fB\-listen\fR
.IP
Accept connections from outside (default: 1 if no \fB\-proxy\fR or \fB\-connect\fR)
.HP
\fB\-listenonion\fR
.IP
Automatically create Tor hidden service (default: 1)
.HP
\fB\-maxconnections=\fR<n>
.IP
Maintain at most <n> connections to peers (temporary service connections
excluded) (default: 125)
.HP
\fB\-maxreceivebuffer=\fR<n>
.IP
Maximum per\-connection receive buffer, <n>*1000 bytes (default: 5000)
.HP
\fB\-maxsendbuffer=\fR<n>
.IP
Maximum per\-connection send buffer, <n>*1000 bytes (default: 1000)
.HP
\fB\-maxtimeadjustment\fR
.IP
Maximum allowed median peer time offset adjustment. Local perspective of
time may be influenced by peers forward or backward by this
amount. (default: 4200 seconds)
.HP
\fB\-maxuploadtarget=\fR<n>
.IP
Tries to keep outbound traffic under the given target (in MiB per 24h),
0 = no limit (default: 0)
.HP
\fB\-onion=\fR<ip:port>
.IP
Use separate SOCKS5 proxy to reach peers via Tor hidden services
(default: \fB\-proxy\fR)
.HP
\fB\-onlynet=\fR<net>
.IP
Make outgoing connections only through network <net> (ipv4, ipv6 or
onion). Incoming connections are not affected by this option.
This option can be specified multiple times to allow multiple
networks.
.HP
\fB\-peerbloomfilters\fR
.IP
Support filtering of blocks and transaction with bloom filters (default:
1)
.HP
\fB\-peertimeout=\fR<n>
.IP
Specify p2p connection timeout in seconds. This option determines the
amount of time a peer may be inactive before the connection to it
is dropped. (minimum: 1, default: 60)
.HP
\fB\-permitbaremultisig\fR
.IP
Relay non\-P2SH multisig (default: 1)
.HP
\fB\-port=\fR<port>
.IP
Listen for connections on <port> (default: 12455 or testnet: 13455)
.HP
\fB\-proxy=\fR<ip:port>
.IP
Connect through SOCKS5 proxy
.HP
\fB\-proxyrandomize\fR
.IP
Randomize credentials for every proxy connection. This enables Tor
stream isolation (default: 1)
.HP
\fB\-seednode=\fR<ip>
.IP
Connect to a node to retrieve peer addresses, and disconnect. This
option can be specified multiple times to connect to multiple
nodes.
.HP
\fB\-socketevents=\fR<mode>
.IP
Socket events mode, which must be one of 'select', 'poll', 'epoll' or
\&'kqueue', depending on your system (default: Linux \- 'epoll',
FreeBSD/Apple \- 'kqueue', Windows \- 'select')
.HP
\fB\-timeout=\fR<n>
.IP
Specify connection timeout in milliseconds (minimum: 1, default: 5000)
.HP
\fB\-torcontrol=\fR<ip>:<port>
.IP
Tor control port to use if onion listening enabled (default:
127.0.0.1:9051)
.HP
\fB\-torpassword=\fR<pass>
.IP
Tor control port password (default: empty)
.HP
\fB\-upnp\fR
.IP
Use UPnP to map the listening port (default: 0)
.HP
\fB\-whitebind=\fR<addr>
.IP
Bind to given address and whitelist peers connecting to it. Use
[host]:port notation for IPv6
.HP
\fB\-whitelist=\fR<IP address or network>
.IP
Whitelist peers connecting from the given IP address (e.g. 1.2.3.4) or
CIDR notated network (e.g. 1.2.3.0/24). Can be specified multiple
times. Whitelisted peers cannot be DoS banned and their
transactions are always relayed, even if they are already in the
mempool, useful e.g. for a gateway
.PP
Indexing options:
.HP
\fB\-addressindex\fR
.IP
Maintain a full address index, used to query for the balance, txids and
unspent outputs for addresses (default: 0)
.HP
\fB\-reindex\fR
.IP
Rebuild chain state and block index from the blk*.dat files on disk
.HP
\fB\-reindex\-chainstate\fR
.IP
Rebuild chain state from the currently indexed blocks
.HP
\fB\-spentindex\fR
.IP
Maintain a full spent index, used to query the spending txid and input
index for an outpoint (default: 0)
.HP
\fB\-timestampindex\fR
.IP
Maintain a timestamp index for block hashes, used to query blocks hashes
by a range of timestamps (default: 0)
.HP
\fB\-txindex\fR
.IP
Maintain a full transaction index, used by the getrawtransaction rpc
call (default: 1)
.PP
Masternode options:
.HP
\fB\-llmq\-data\-recovery=\fR<n>
.IP
Enable automated quorum data recovery (default: 1)
.HP
\fB\-llmq\-qvvec\-sync=\fR<quorum_name>:<mode>
.IP
Defines from which LLMQ type the masternode should sync quorum
verification vectors. Can be used multiple times with different
LLMQ types. <mode>: 0 (sync always from all quorums of the type
defined by <quorum_name>), 1 (sync from all quorums of the type
defined by <quorum_name> if a member of any of the quorums)
.HP
\fB\-masternodeblsprivkey=\fR<hex>
.IP
Set the masternode BLS private key and enable the client to act as a
masternode
.HP
\fB\-platform\-user=\fR<user>
.IP
Set the username for the "platform user", a restricted user intended to
be used by GoByte Platform, to the specified username.
.PP
Statsd options:
.HP
\fB\-statsenabled\fR
.IP
Publish internal stats to statsd (default: 0)
.HP
\fB\-statshost=\fR<ip>
.IP
Specify statsd host (default: 127.0.0.1)
.HP
\fB\-statshostname=\fR<ip>
.IP
Specify statsd host name (default: )
.HP
\fB\-statsns=\fR<ns>
.IP
Specify additional namespace prefix (default: )
.HP
\fB\-statsperiod=\fR<seconds>
.IP
Specify the number of seconds between periodic measurements (default:
60)
.HP
\fB\-statsport=\fR<port>
.IP
Specify statsd port (default: 8125)
.PP
Wallet options:
.HP
\fB\-createwalletbackups=\fR<n>
.IP
Number of automatic wallet backups (default: 10)
.HP
\fB\-disablewallet\fR
.IP
Do not load the wallet and disable wallet RPC calls
.HP
\fB\-instantsendnotify=\fR<cmd>
.IP
Execute command when a wallet InstantSend transaction is successfully
locked (%s in cmd is replaced by TxID)
.HP
\fB\-keypool=\fR<n>
.IP
Set key pool size to <n> (default: 1000)
.HP
\fB\-rescan=\fR<mode>
.IP
Rescan the block chain for missing wallet transactions on startup (1 =
start from wallet creation time, 2 = start from genesis block)
.HP
\fB\-salvagewallet\fR
.IP
Attempt to recover private keys from a corrupt wallet on startup
.HP
\fB\-spendzeroconfchange\fR
.IP
Spend unconfirmed change when sending transactions (default: 1)
.HP
\fB\-upgradewallet\fR
.IP
Upgrade wallet to latest format on startup
.HP
\fB\-wallet=\fR<path>
.IP
Specify wallet database path. Can be specified multiple times to load
multiple wallets. Path is interpreted relative to <walletdir> if
it is not absolute, and will be created if it does not exist (as
a directory containing a wallet.dat file and log files). For
backwards compatibility this will also accept names of existing
data files in <walletdir>.)
.HP
\fB\-walletbackupsdir=\fR<dir>
.IP
Specify full path to directory for automatic wallet backups (must exist)
.HP
\fB\-walletbroadcast\fR
.IP
Make the wallet broadcast transactions (default: 1)
.HP
\fB\-walletdir=\fR<dir>
.IP
Specify directory to hold wallets (default: <datadir>/wallets if it
exists, otherwise <datadir>)
.HP
\fB\-walletnotify=\fR<cmd>
.IP
Execute command when a wallet transaction changes (%s in cmd is replaced
by TxID)
.HP
\fB\-zapwallettxes=\fR<mode>
.IP
Delete all wallet transactions and only recover those parts of the
blockchain through \fB\-rescan\fR on startup (1 = keep tx meta data e.g.
account owner and payment request information, 2 = drop tx meta
data)
.HP
\fB\-discardfee=\fR<amt>
.IP
The fee rate (in GBX/kB) that indicates your tolerance for discarding
change by adding it to the fee (default: 0.0001). Note: An output
is discarded if it is dust at this rate, but we will always
discard up to the dust relay fee and a discard fee above that is
limited by the fee estimate for the longest target
.HP
\fB\-fallbackfee=\fR<amt>
.IP
A fee rate (in GBX/kB) that will be used when fee estimation has
insufficient data (default: 0.00001)
.HP
\fB\-mintxfee=\fR<amt>
.IP
Fees (in GBX/kB) smaller than this are considered zero fee for
transaction creation (default: 0.00001)
.HP
\fB\-paytxfee=\fR<amt>
.IP
Fee (in GBX/kB) to add to transactions you send (default: 0.00)
.HP
\fB\-txconfirmtarget=\fR<n>
.IP
If paytxfee is not set, include enough fee so transactions begin
confirmation on average within n blocks (default: 6)
.HP
\fB\-hdseed=\fR<hex>
.IP
User defined seed for HD wallet (should be in hex). Only has effect
during wallet creation/first start (default: randomly generated)
.HP
\fB\-mnemonic=\fR<text>
.IP
User defined mnemonic for HD wallet (bip39). Only has effect during
wallet creation/first start (default: randomly generated)
.HP
\fB\-mnemonicpassphrase=\fR<text>
.IP
User defined mnemonic passphrase for HD wallet (BIP39). Only has effect
during wallet creation/first start (default: empty string)
.HP
\fB\-usehd\fR
.IP
Use hierarchical deterministic key generation (HD) after BIP39/BIP44.
Only has effect during wallet creation/first start (default: 0)
.HP
\fB\-keepass\fR
.IP
Use KeePass 2 integration using KeePassHttp plugin (default: 0)
.HP
\fB\-keepassid=\fR<id>
.IP
KeePassHttp id for the established association
.HP
\fB\-keepasskey=\fR<key>
.IP
KeePassHttp key for AES encrypted communication with KeePass
.HP
\fB\-keepassname=\fR<name>
.IP
Name to construct url for KeePass entry that stores the wallet
passphrase
.HP
\fB\-keepassport=\fR<port>
.IP
Connect to KeePassHttp on port <port> (default: 19455)
.HP
\fB\-coinjoinamount=\fR<n>
.IP
Target CoinJoin balance (2\-21000000, default: 1000)
.HP
\fB\-coinjoinautostart\fR
.IP
Start CoinJoin automatically (0\-1, default: 0)
.HP
\fB\-coinjoindenomsgoal=\fR<n>
.IP
Try to create at least N inputs of each denominated amount (10\-100000,
default: 50)
.HP
\fB\-coinjoindenomshardcap=\fR<n>
.IP
Create up to N inputs of each denominated amount (10\-100000, default:
300)
.HP
\fB\-coinjoinmultisession\fR
.IP
Enable multiple CoinJoin mixing sessions per block, experimental (0\-1,
default: 0)
.HP
\fB\-coinjoinrounds=\fR<n>
.IP
Use N separate masternodes for each denominated input to mix funds
(2\-16, default: 4)
.HP
\fB\-coinjoinsessions=\fR<n>
.IP
Use N separate masternodes in parallel to mix funds (1\-10, default: 4)
.HP
\fB\-enablecoinjoin\fR
.IP
Enable use of CoinJoin for funds stored in this wallet (0\-1, default: 0)
.PP
ZeroMQ notification options:
.HP
\fB\-zmqpubhashblock=\fR<address>
.IP
Enable publish hash block in <address>
.HP
\fB\-zmqpubhashgovernanceobject=\fR<address>
.IP
Enable publish hash of governance objects (like proposals) in <address>
.HP
\fB\-zmqpubhashgovernancevote=\fR<address>
.IP
Enable publish hash of governance votes in <address>
.HP
\fB\-zmqpubhashinstantsenddoublespend=\fR<address>
.IP
Enable publish transaction hashes of attempted InstantSend double spend
in <address>
.HP
\fB\-zmqpubhashrecoveredsig=\fR<address>
.IP
Enable publish message hash of recovered signatures (recovered by LLMQs)
in <address>
.HP
\fB\-zmqpubhashtx=\fR<address>
.IP
Enable publish hash transaction in <address>
.HP
\fB\-zmqpubhashtxlock=\fR<address>
.IP
Enable publish hash transaction (locked via InstantSend) in <address>
.HP
\fB\-zmqpubrawblock=\fR<address>
.IP
Enable publish raw block in <address>
.HP
\fB\-zmqpubrawinstantsenddoublespend=\fR<address>
.IP
Enable publish raw transactions of attempted InstantSend double spend in
<address>
.HP
\fB\-zmqpubrawrecoveredsig=\fR<address>
.IP
Enable publish raw recovered signatures (recovered by LLMQs) in
<address>
.HP
\fB\-zmqpubrawtx=\fR<address>
.IP
Enable publish raw transaction in <address>
.HP
\fB\-zmqpubrawtxlock=\fR<address>
.IP
Enable publish raw transaction (locked via InstantSend) in <address>
.PP
Debugging/Testing options:
.HP
\fB\-debug=\fR<category>
.IP
Output debugging information (default: 0, supplying <category> is
optional). If <category> is not supplied or if <category> = 1,
output all debugging information. <category> can be: net, tor,
mempool, http, bench, zmq, db, rpc, estimatefee, addrman,
selectcoins, reindex, cmpctblock, rand, prune, proxy, mempoolrej,
libevent, coindb, qt, leveldb, chainlocks, gobject, instantsend,
keepass, llmq, llmq\-dkg, llmq\-sigs, mnpayments, mnsync, coinjoin,
spork, netconn.
.HP
\fB\-debugexclude=\fR<category>
.IP
Exclude debugging information for a category. Can be used in conjunction
with \fB\-debug\fR=\fI\,1\/\fR to output debug logs for all categories except one
or more specified categories.
.HP
\fB\-disablegovernance\fR
.IP
Disable governance validation (0\-1, default: 0)
.HP
\fB\-help\-debug\fR
.IP
Show all debugging options (usage: \fB\-\-help\fR \fB\-help\-debug\fR)
.HP
\fB\-highsubsidyblocks=\fR<n>
.IP
The number of blocks with a higher than normal subsidy to mine at the
start of a devnet (default: 0)
.HP
\fB\-highsubsidyfactor=\fR<n>
.IP
The factor to multiply the normal block subsidy by while in the
highsubsidyblocks window of a devnet (default: 1)
.HP
\fB\-llmqchainlocks=\fR<quorum name>
.IP
Override the default LLMQ type used for ChainLocks on a devnet. Allows
using ChainLocks with smaller LLMQs. (default: llmq_50_60)
.HP
\fB\-llmqdevnetparams=\fR<size:threshold>
.IP
Override the default LLMQ size for the LLMQ_DEVNET quorum (default:
10:6)
.HP
\fB\-llmqinstantsend=\fR<quorum name>
.IP
Override the default LLMQ type used for InstantSend on a devnet. Allows
using InstantSend with smaller LLMQs. (default: llmq_50_60)
.HP
\fB\-llmqtestparams=\fR<size:threshold>
.IP
Override the default LLMQ size for the LLMQ_TEST quorum (default: 3:2)
.HP
\fB\-logips\fR
.IP
Include IP addresses in debug output (default: 0)
.HP
\fB\-logtimestamps\fR
.IP
Prepend debug output with timestamp (default: 1)
.HP
\fB\-maxtxfee=\fR<amt>
.IP
Maximum total fees (in GBX) to use in a single wallet transaction or
raw transaction; setting this too low may abort large
transactions (default: 0.10)
.HP
\fB\-minimumdifficultyblocks=\fR<n>
.IP
The number of blocks that can be mined with the minimum difficulty at
the start of a devnet (default: 0)
.HP
\fB\-minsporkkeys=\fR<n>
.IP
Overrides minimum spork signers to change spork value. Only useful for
regtest and devnet. Using this on mainnet or testnet will ban
you.
.HP
\fB\-printtoconsole\fR
.IP
Send trace/debug info to console instead of debug.log file
.HP
\fB\-shrinkdebugfile\fR
.IP
Shrink debug.log file on client startup (default: 1 when no \fB\-debug\fR)
.HP
\fB\-sporkaddr=\fR<gobyteaddress>
.IP
Override spork address. Only useful for regtest and devnet. Using this
on mainnet or testnet will ban you.
.HP
\fB\-sporkkey=\fR<privatekey>
.IP
Set the private key to be used for signing spork messages.
.HP
\fB\-uacomment=\fR<cmt>
.IP
Append comment to the user agent string
.PP
Chain selection options:
.HP
\fB\-devnet=\fR<name>
.IP
Use devnet chain with provided name
.HP
\fB\-testnet\fR
.IP
Use the test chain
.PP
Node relay options:
.HP
\fB\-bytespersigop\fR
.IP
Minimum bytes per sigop in transactions we relay and mine (default: 20)
.HP
\fB\-datacarrier\fR
.IP
Relay and mine data carrier transactions (default: 1)
.HP
\fB\-datacarriersize\fR
.IP
Maximum size of data in data carrier transactions we relay and mine
(default: 83)
.HP
\fB\-minrelaytxfee=\fR<amt>
.IP
Fees (in GBX/kB) smaller than this are considered zero fee for
relaying, mining and transaction creation (default: 0.00001)
.HP
\fB\-whitelistforcerelay\fR
.IP
Force relay of transactions from whitelisted peers even if they violate
local relay policy (default: 1)
.HP
\fB\-whitelistrelay\fR
.IP
Accept relayed transactions received from whitelisted peers even when
not relaying transactions (default: 1)
.PP

doc/release-notes.md
GoByte Core version 0.17.0.3
==========================

Release is now available from:

  <https://www.gobyte.network/downloads/#wallets>

This is a new major version release, bringing new features, various bugfixes
and other improvements.

This release is mandatory for all nodes.

Please report bugs using the issue tracker at github:

  <https://github.com/gobytecoin/gobyte/issues>


Upgrading and downgrading
=========================

How to Upgrade
--------------

If you are running an older version, shut it down. Wait until it has completely
shut down (which might take a few minutes for older versions), then run the
installer (on Windows) or just copy over /Applications/GoByte-Qt (on Mac) or
gobyted/gobyte-qt (on Linux). If you upgrade after DIP0003 activation and you were
using version < 0.13 you will have to reindex (start with -reindex-chainstate
or -reindex) to make sure your wallet has all the new data synced. Upgrading
from version 0.13 should not require any additional actions.

When upgrading from a version prior to 0.14.0.3, the
first startup of GoByte Core will run a migration process which can take a few
minutes to finish. After the migration, a downgrade to an older version is only
possible with a reindex (or reindex-chainstate).

Downgrade warning
-----------------

### Downgrade to a version < 0.14.0.3

Downgrading to a version older than 0.14.0.3 is no longer supported due to
changes in the "evodb" database format. If you need to use an older version,
you must either reindex or re-sync the whole chain.

### Downgrade of masternodes to < 0.17.0.3

Starting with the 0.16 release, masternodes verify the protocol version of other
masternodes. This results in PoSe punishment/banning for outdated masternodes,
so downgrading even prior to the activation of the introduced hard-fork changes
is not recommended.

Notable changes
===============

Opcode updates
--------------
Several opcodes have been reactivated/introduced to broaden the functionality
of the system and enable developers to build new solutions. These opcodes are
a combination of previously disabled ones that have been found to be safe and
new ones previously introduced by Bitcoin Cash. Details of the opcodes are
provided in [DIP-0020](https://github.com/dashpay/dips/blob/master/dip-0020.md).

These opcodes are activated via a BIP9 style hard fork that will begin
signalling on July 1st using bit 6. Any nodes that do not upgrade by the time
this feature is activated will diverge from the rest of the network.

DKG Data Sharing
----------------
Quorum resilience has been improved by enabling masternodes to request DKG data
from other quorum members. This allows GoByte Platform to obtain required
information while also making it possible for corrupted masternodes to recover
the DKG data they need to participate in quorums they are part of. Details are
provided in [DIP-0021](https://github.com/dashpay/dips/blob/master/dip-0021.md).

Platform support
----------------
Support for GoByte Platform has been expanded through the addition of a new
quorum type `LLMQ_100_67`, several RPCs, and a way to limit Platform RPC access
to a subset of allowed RPCs, specifically:
- `getbestblockhash`
- `getbestchainlock`
- `getblockcount`
- `getblockhash`
- `quorum sign` (for platform quorums only)
- `quorum verify`
- `verifyislock`

These changes provide necessary Platform capabilities while maximizing the
isolation between Core and Platform. New quorum type will be activated using
the same bit 6 introduced to activate new opcodes.

BLS update
----------
GoByte Core’s BLS signature library has been updated based on v1.0 of the
Chia BLS library to support migration to a new BLS signature scheme which will
be implemented in a future version of GoByteCore. These changes will be made to
align with standards and improve security.

Network performance improvements
--------------------------------
This version of GoByte Core includes multiple optimizations to the network and
p2p message handling code.

We reintroduced [Intra-Quorum Connections](https://github.com/dashpay/dips/blob/master/dip-0006.md#intra-quorum-communication)
which were temporary disabled with the introduction of
`SPORK_21_QUORUM_ALL_CONNECTED`. This should make communications for masternodes
belonging to the same quorum more robust and improve network connectivity in
general.

FreeBSD and MacOS machines will now default to the new `kqueue` network mode
which is similar to `epoll` mode on linux-based. It removes most of the CPU
overhead caused by the sub-optimal use of `select` in the network thread when
many connections were involved.

Wallet changes
--------------
Upgrading a non-Hierarchical Deterministic (HD) wallet to an HD wallet is now
possible. Previously new HD wallets could be created, but non-HD wallets could
not be upgraded. This update will enable existing non-HD wallets to upgrade to
take advantage of HD wallet features. Upgrades can be done via either the debug
console or command line and a new backup must be made when this upgrade is
performed.

Users can now load and unload wallets dynamically via RPC and GUI console.
It's now possible to have multiple wallets loaded at the same time, however only
one of them is going to be active and receive updates or be used for sending
transactions at a specific point in time.

Also, enabling wallet encryption no longer requires a wallet restart.

Sporks
------
Several spork changes have been made to streamline code and improve system
reliability. Activation of `SPORK_22_PS_MORE_PARTICIPANTS` in GoByteCore v0.16
has rendered that spork unnecessary. The associated logic has been hardened
and the spork removed. `SPORK_21_QUORUM_ALL_CONNECTED` logic has been split
into two sporks, `SPORK_21_QUORUM_ALL_CONNECTED` and `SPORK_23_QUORUM_POSE`,
so that masternode quorum connectivity and quorum Proof of Service (PoSe) can
be controlled independently. Finally, `SPORK_2_INSTANTSEND_ENABLED` has a new

src/gobyted.cpp
// Copyright (c) 2009-2010 Satoshi Nakamoto
// Copyright (c) 2009-2015 The Bitcoin Core developers
// Copyright (c) 2014-2021 The Dash Core developers
// Copyright (c) 2026-2026 The GoByte Core developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#if defined(HAVE_CONFIG_H)
#include <config/gobyte-config.h>
#endif

#include <chainparams.h>
#include <clientversion.h>
#include <compat.h>
#include <fs.h>
#include <httprpc.h>
#include <httpserver.h>
#include <init.h>
#include <noui.h>
#include <rpc/server.h>
#include <stacktraces.h>
#include <util.h>
#include <utilstrencodings.h>
#include <walletinitinterface.h>

#include <stdio.h>

/* Introduction text for doxygen: */

/*! \mainpage Developer documentation
 *
 * \section intro_sec Introduction
 *
 * This is the developer documentation of the reference client for an experimental new digital currency called GoByte (https://www.gobyte.network/),
 * which enables instant payments to anyone, anywhere in the world. GoByte uses peer-to-peer technology to operate
 * with no central authority: managing transactions and issuing money are carried out collectively by the network.
 *
 * The software is a community-driven open source project, released under the MIT license.
 *
 * \section Navigation
 * Use the buttons <code>Namespaces</code>, <code>Classes</code> or <code>Files</code> at the top of the page to start navigating the code.
 */

void WaitForShutdown()
{
    while (!ShutdownRequested()) {
        MilliSleep(200);
    }
    Interrupt();
}

//////////////////////////////////////////////////////////////////////////////
//
// Start
//
bool AppInit(int argc, char* argv[])
{
    bool fRet = false;

    //
    // Parameters
    //
    // If Qt is used, parameters/gobyte.conf are parsed in qt/gobyte.cpp's main()
    SetupServerArgs();
#if HAVE_DECL_DAEMON
    gArgs.AddArg("-daemon", "Run in the background as a daemon and accept commands", false, OptionsCategory::OPTIONS);
#endif
    gArgs.ParseParameters(argc, argv);

    if (gArgs.IsArgSet("-printcrashinfo")) {
        std::cout << GetCrashInfoStrFromSerializedStr(gArgs.GetArg("-printcrashinfo", "")) << std::endl;
        return true;
    }

    // Process help and version before taking care about datadir
    if (gArgs.IsArgSet("-?") || gArgs.IsArgSet("-h") || gArgs.IsArgSet("-help") || gArgs.IsArgSet("-version")) {
        std::string strUsage = strprintf("%s Daemon", PACKAGE_NAME) + " version " + FormatFullVersion() + "\n";

        if (gArgs.IsArgSet("-version")) {
            strUsage += FormatParagraph(LicenseInfo());
        } else {
            strUsage += "\nUsage:\n"
                        "  gobyted [options]                     " +
                        strprintf("Start %s Daemon", PACKAGE_NAME) + "\n";

            strUsage += "\n" + gArgs.GetHelpMessage();
        }

        fprintf(stdout, "%s", strUsage.c_str());
        return true;
    }

    try {
        bool datadirFromCmdLine = gArgs.IsArgSet("-datadir");
        if (datadirFromCmdLine && !fs::is_directory(GetDataDir(false))) {
            fprintf(stderr, "Error: Specified data directory \"%s\" does not exist.\n", gArgs.GetArg("-datadir", "").c_str());
            return false;
        }
        try {
            gArgs.ReadConfigFile(gArgs.GetArg("-conf", BITCOIN_CONF_FILENAME));
        } catch (const std::exception& e) {
            fprintf(stderr, "Error reading configuration file: %s\n", e.what());
            return false;
        }
        if (!datadirFromCmdLine && !fs::is_directory(GetDataDir(false))) {
            fprintf(stderr, "Error: Specified data directory \"%s\" from config file does not exist.\n", gArgs.GetArg("-datadir", "").c_str());
            return EXIT_FAILURE;
        }
        // Check for -testnet or -regtest parameter (Params() calls are only valid after this clause)
        try {
            SelectParams(gArgs.GetChainName());
        } catch (const std::exception& e) {
            fprintf(stderr, "Error: %s\n", e.what());
            return false;
        }

        // Error out when loose non-argument tokens are encountered on command line
        for (int i = 1; i < argc; i++) {
            if (!IsSwitchChar(argv[i][0])) {
                fprintf(stderr, "Error: Command line contains unexpected token '%s', see gobyted -h for a list of options.\n", argv[i]);
                return false;
            }
        }

        // -server defaults to true for gobyted but not for the GUI so do this here
        gArgs.SoftSetBoolArg("-server", true);
        // Set this early so that parameter interactions go to console
        InitLogging();
        InitParameterInteraction();
        if (!AppInitBasicSetup()) {
            // InitError will have been called with detailed error, which ends up on console
            return false;
        }
        if (!AppInitParameterInteraction()) {
            // InitError will have been called with detailed error, which ends up on console
            return false;
        }
        if (!AppInitSanityChecks()) {
            // InitError will have been called with detailed error, which ends up on console
            return false;
        }
        if (gArgs.GetBoolArg("-daemon", false)) {
#if HAVE_DECL_DAEMON
#if defined(MAC_OSX)
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
#endif
            fprintf(stdout, "GoByte Core server starting\n");

            // Daemonize
            if (daemon(1, 0)) { // don't chdir (1), do close FDs (0)
                fprintf(stderr, "Error: daemon() failed: %s\n", strerror(errno));
                return false;
            }
#if defined(MAC_OSX)
#pragma GCC diagnostic pop
#endif
#else
            fprintf(stderr, "Error: -daemon is not supported on this operating system\n");
            return false;
#endif // HAVE_DECL_DAEMON
        }
        // Lock data directory after daemonization
        if (!AppInitLockDataDirectory()) {
            // If locking the data directory failed, exit immediately
            return false;
        }
        fRet = AppInitMain();
    } catch (...) {
        PrintExceptionContinue(std::current_exception(), "AppInit()");
    }

    if (!fRet) {
        Interrupt();
    } else {
        WaitForShutdown();
    }
    Shutdown();

    return fRet;
}

int main(int argc, char* argv[])
{
    RegisterPrettyTerminateHander();
    RegisterPrettySignalHandlers();

    SetupEnvironment();

    // Connect gobyted signal handlers
    noui_connect();

    return (AppInit(argc, argv) ? EXIT_SUCCESS : EXIT_FAILURE);
}
